src/evict/evict_page.c

```diff
@@ -143,9 +144,22 @@ done:	if (((inmem_split && ret == 0) || (forced_eviction && ret == EBUSY)) &&
  * __wt_evict_page_clean_update --
  *	Update a clean page's reference on eviction.
  */
-void
-__wt_evict_page_clean_update(WT_SESSION_IMPL *session, WT_REF *ref)
+int
+__wt_evict_page_clean_update(
+    WT_SESSION_IMPL *session, WT_REF *ref, int exclusive)
 {
+	int evict;
+
+	/*
+	 * If doing normal system eviction, but only in the service of reducing
+	 * the number of dirty pages, leave the clean page in cache.
+	 */
+	if (!exclusive) {
```

Comment:
I don't really follow how the exclusive flag maps to "normal system eviction". Can you make that cleaner?